docs/docs/sample-scripts/aad/analyze-user-profile-photos.md

```diff
@@ -0,0 +1,255 @@
+$outputDir = "$executionDir/$resultDir"
+$outputFilePath = "$outputDir/$(get-date -f yyyyMMdd-HHmmss)-scan-profile-pictures-outcome.csv"
+
+if (-not (Test-Path -Path "$outputDir" -PathType Container)) {
+    Write-Host "Creating $outputDir folder..."
+    New-Item -ItemType Directory -Path "$outputDir"
+}
+
+function AddAnalysisOutcome {
+    param (
+        [Parameter(Mandatory = $false)] [string] $UserId,
+        [Parameter(Mandatory = $false)] [string] $UserMail,
+        [Parameter(Mandatory = $false)] [bool] $IsPortraitValid,
+        [Parameter(Mandatory = $false)] [bool] $IsOnlyOnePersonValid,
+        [Parameter(Mandatory = $false)] [bool] $IsClipartValid,
+        [Parameter(Mandatory = $false)] [bool] $IsLineDrawingValid,
+        [Parameter(Mandatory = $false)] [bool] $IsAdultValid,
+        [Parameter(Mandatory = $false)] [bool] $IsRacyValid,
+        [Parameter(Mandatory = $false)] [bool] $IsGoryValid,
+        [Parameter(Mandatory = $false)] [bool] $IsCelebrity,
+        [Parameter(Mandatory = $false)] [bool] $IsForbiddenKeywordExist,
+        [Parameter(Mandatory = $false)] [bool] $IsValidProfilePhoto,
+        [Parameter(Mandatory = $false)] [string] $Notes
+    )
+
+    $analysisOutcome = New-Object -TypeName PSObject
+
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "UserId" -Value $UserId
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "UserMail" -Value $UserMail
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "IsPortraitValid" -Value $IsPortraitValid
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "IsOnlyOnePersonValid" -Value $IsOnlyOnePersonValid
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "IsClipartValid" -Value $IsClipartValid
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "IsLineDrawingValid" -Value $IsLineDrawingValid
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "IsAdultValid" -Value $IsAdultValid
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "IsRacyValid" -Value $IsRacyValid
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "IsGoryValid" -Value $IsGoryValid
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "IsCelebrity" -Value $IsCelebrity
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "IsForbiddenKeywordExist" -Value $IsForbiddenKeywordExist
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "IsValidProfilePhoto" -Value $IsValidProfilePhoto
+    $analysisOutcome | Add-Member -MemberType NoteProperty -Name "Notes" -Value $Notes
+
+    $global:analysisOutcomes += $analysisOutcome
+}
+
+$users = m365 aad user list --properties $requiredProfileProperties -o json | ConvertFrom-Json -AsHashtable
+
+foreach ($user in $users) {
+
+    try {
+        $userId = $user.id
+        $userMail = $user.mail
+
+        try {
+            $token = m365 util accesstoken get --resource https://graph.microsoft.com --new
```

Comment:
We shouldn't need to retrieve a new token for each user request. If the token expired or is about to expire, CLI will automatically retrieve a new token.

Reply:
Sure @waldekmastykarz. I have moved this logic outside the foreach loop. We are good with this now.